plugin/helpers_channels.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugin
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/utils"
+	"github.com/pkg/errors"
+)
+
+func (p *HelpersImpl) EnsureChannel(channel *model.Channel) (retChannelId string, retErr error) {
+	// Must provide a channel with a name and teadId
+	if channel == nil || len(channel.Name) < 1 || len(channel.TeamId) < 1 {
+		return "", errors.New("passed a bad channel, nil or no name or no team id")
+	}
+
+	// If we fail for any reason, this could be a race between creation of channel and
+	// retrieval from another EnsureChannel. Just try the basic retrieve existing again.
+	defer func() {
+		if retChannelId == "" || retErr != nil {
+			var err error
+			var channelIdBytes []byte
+
+			err = utils.ProgressiveRetry(func() error {
+				channelIdBytes, err = p.API.KVGet(CHANNEL_KEY)
+				if err != nil {
+					return err
+				}
+				return nil
+			})
+
+			if err == nil && channelIdBytes != nil {
+				retChannelId = string(channelIdBytes)
+				retErr = nil
+			}
+		}
+	}()
+
+	// Fetch channel ID from key value store
+	channelIdBytes, kvGetErr := p.API.KVGet(CHANNEL_KEY)
+	if kvGetErr != nil {
+		// Failed to retrive the value of channel
+		return "", errors.Wrap(kvGetErr, "failed to get channel ID")
+	}
+
+	// Check if Channel ID exists
+	if channelIdBytes != nil {
+		channelId := string(channelIdBytes)
+		return channelId, nil
```

Comment:
We don't update the metadata if the channel already exists?

Reply:
Yes It should update the metadata if channel exists. I will fix it.